src/overlays/actors/ovl_Bg_Spot16_Bombstone/z_bg_spot16_bombstone.c

```diff
@@ -21,49 +94,434 @@ const ActorInit Bg_Spot16_Bombstone_InitVars = {
+    f32 velocityYMultiplier = 1.3f;
+    Vec3f randomPosition;
+    Vec3f randomVelocity;
+    s32 index;
+    s16 scale;
+
+    if (this->actor.params == 0) {
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_SPOT16_BOMBSTONE, this->actor.posRot.pos.x,
+                    this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 5);
+        index = 3;
+    } else {
+        index = 0;
+    }
+
+    if (index < ARRAY_COUNT(D_808B6074)) {
+        do {
+            randomPosition.x = ((Math_Rand_ZeroOne() - 0.5f) * 8.0f) + this->actor.posRot.pos.x;
+            randomPosition.y = ((Math_Rand_ZeroOne() * 5.0f) + this->actor.posRot.pos.y) + 8.0f;
+            randomPosition.z = ((Math_Rand_ZeroOne() - 0.5f) * 8.0f) + this->actor.posRot.pos.z;
+
+            randomVelocity.x = (Math_Rand_ZeroOne() - 0.5f) * 16.0f;
+            randomVelocity.y = (Math_Rand_ZeroOne() * 14.0) + (fabsf(this->actor.velocity.y) * velocityYMultiplier);
+            randomVelocity.z = (Math_Rand_ZeroOne() - 0.5f) * 16.0f;
+
+            scale = D_808B6074[index] * this->actor.scale.x * 3;
+
+            func_80029E8C(globalCtx, &randomPosition, &randomVelocity, &this->actor.posRot.pos, -0x1A4, 0x31, 0xF, 0xF,
+                          0, scale, 2, 0x40, 0xA0, -1, OBJECT_BOMBIWA, D_060009E0);
+            index += 1;
+        } while (index != ARRAY_COUNT(D_808B6074));
+    }
+}
+
+void func_808B561C(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    s32 index;
+    PosRot* posRot;
+
+    posRot = &this->actor.posRot;
+    for (index = 0; index < ARRAY_COUNT(D_808B6088); index++) {
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_SPOT16_BOMBSTONE, posRot->pos.x, posRot->pos.y,
+                        posRot->pos.z, 0, 0, 0, D_808B6088[index]) == NULL) {
+            break;
+        }
+    }
+}
+
+void func_808B56BC(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 sinValue;
+    s16 adjustedYawDiff;
+    s32 yawDiff;
+    s32 absYawDiff;
+
+    if (this->actor.xzDistFromLink < 130.0f && this->actor.yDistFromLink < 160.0f &&
+        this->actor.yDistFromLink >= -10.0f) {
+        yawDiff = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+        absYawDiff = ABS(yawDiff);
+
+        adjustedYawDiff = absYawDiff - 0x3FFF;
+
+        if (adjustedYawDiff > 0) {
+            sinValue = Math_Sins(adjustedYawDiff) * this->actor.xzDistFromLink;
+
+            if (sinValue >= 0.0f) {
+                player->actor.posRot.pos.x += sinValue * this->sinRotation;
+                player->actor.posRot.pos.z += sinValue * this->cosRotation;
+            } else {
+                osSyncPrintf("Error 補正出来ない(%s %d)(arg_data 0x%04x)(hosei_angY %x)\n",
+                             "../z_bg_spot16_bombstone.c", 935, this->actor.params, adjustedYawDiff);
+            }
+        }
+    }
+}
+
+void func_808B57E0(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    EnBombf* playerHeldActor;
```

Comment:
Id personally type this as just Actor*, it checks if its a bombf after the fact, but the point of the temp is to hold whatever actor player is currently holding